thunder/distributed/__init__.py

```diff
@@ -499,7 +499,9 @@ def fsdp_transform_module(
         # This is done before sharding in case the materialization logic depends on the tensor shape.
         # The tradeoff is that all of a module's direct parameters need to fit in device.
         # Each module only initializes its own parameters and not those of its children (recurse=False)
-        if any(t.is_meta for t in chain(module_copy.parameters(recurse=False), module_copy.buffers(recurse=False))):
+        if is_meta_initialized := any(
```

Comment:
I was confused by this name because "meta initialization" should not use any device memory.
```suggestion
        if is_meta_materialized := any(
```